tests/rptest/services/redpanda.py

```diff
@@ -911,7 +911,7 @@ def raise_on_crash(self):
             if crash_log:
                 crashes.append((node, line))
 
-        if not crashes:
+        if not crashes and not allow_missing_process:
```

Comment:
This makes sense, I am working on adding changes so that system state can optionally be restored on context exit. The actions will need to decide how they want to restore the changes made. I will add the changes to this same PR and remove this new flag.

Reply:
I have added changes to restore the state now, and also removed the `allow_missing_process` flag